test/test_package.cpp

```diff
@@ -1786,3 +1786,21 @@ TEST(DDPackageTest, TwoQubitGateCreationFailure) {
 
     EXPECT_THROW(dd->makeTwoQubitGateDD(dd::CXmat, 2, 0, 1), std::runtime_error);
 }
+
+TEST(DDPackageTest, InnerProductTopNodeConjugation) {
+    // Test comes from experimental results
+    // 2 qubit state is rotated Rxx(-2) equivalent to
+    // Ising model evolution up to a time T=1
+    const dd::QubitCount nrQubits  = 2;
+    const auto           dd        = std::make_unique<dd::Package<>>(nrQubits);
+    const auto           zeroState = dd->makeZeroState(nrQubits);
+    const auto           rxx       = dd->makeRXXDD(nrQubits, 0, 1, -2);
+    const auto           op        = dd->makeGateDD(dd::Zmat, nrQubits, 0);
+
+    const auto evolvedState = dd->multiply(rxx, zeroState);
+
+    // Actual evolution results in approximately -0.4161468365471423
+    // If the top node in the inner product is not conjugated properly,
+    // it will result in +0.4161468365471423.
+    EXPECT_NEAR(dd->expectationValue(op, evolvedState), -0.4161468365471423, 0.001);
```

Comment:
Just a very small comment: wouldn't it be sufficient to make sure the sign of the expectation value is correct?

Reply:
That was what I originally did, but I think it might be too imprecise. We could easily have another bug where it's less than 0 but still not correct. I tried finding a cleaner number (like a rotation that results in 1), but the values are never super nice.

I would leave it as this one so we know it's 100% being calculated properly, but it's up to you.

Reply:
If it looks nicer, we could round the number to like 0.416 or 0.42 even and that would be enough

Reply:
I do like that second suggestion. Then the number doesn't look that unwieldy and it matches the significant digits of the tolerance being used.
0.416 sounds good to me.

Reply:
Agreed, done.